### DISCUSSION TITLE
Preview for VS Code projects (*.code-workspace)

### DISCUSSION THREAD

**Discussion by maxx27** [sbarex/SourceCodeSyntaxHighlight#251] [Q&A] [answered]
I want to enable preview for VS Code Projects: `*.code-workspace` files. They are in JSON with comments format.

`file` recognised as JSON data:

```console
$ file MyProject.code-workspace
MyProject.code-workspace: JSON data
```

I tried different options:

![image](https://github.com/sbarex/SourceCodeSyntaxHighlight/assets/5478638/6fa43c3b-3c9c-4df4-a30f-2702e7e35f16)

![image](https://github.com/sbarex/SourceCodeSyntaxHighlight/assets/5478638/178f49e7-9a64-41d0-a269-0db1bde70a54)

But it doesn't help. In preview I see the following:

![image](https://github.com/sbarex/SourceCodeSyntaxHighlight/assets/5478638/1e5213fc-f11b-4d92-98bf-3dfbcdb91efc)

I read that

> JSON files (.json, .jsonc) ** On macOS 13 Ventura with Apple Silicon the .json extension is reserved by the system and cannot be handled.**

But I don't use a `*.json` file extension.

Could you please help me to configure syntax-highlight?

**Comment by sbarex** [comment, marked as answer]
You cannot manual handle a custom file extension. Only file *without* an extension!
I can add the `.code-workspace` extension support on the next release.

**Reply by maxx27** [reply]
Ouch! I got it now:

<img src="https://github.com/sbarex/SourceCodeSyntaxHighlight/assets/5478638/f63e9037-1a2f-4d0d-96a7-6ab779e994f4" width=50% height=50%>


> I can add the .code-workspace extension support on the next release.

Yes, it would be nice!

Is it possible to create inquiry file and add new formats without new release? For example, in case when I will have another file format.

**Reply by sbarex** [reply]
Apple does not allow you to customize the list of managed files at runtime, but requires you to recompile the app with the updated plist.
